kernel_hardening_checker/__init__.py

```diff
@@ -48,13 +48,34 @@ def detect_arch(fname: str, supported_archs: List[str]) -> Tuple[StrOrNone, str]
                 if arch is None:
                     arch = option
                 else:
-                    return None, 'detected more than one microarchitecture'
+                    return None, 'detected more than one microarchitecture in kconfig'
 
     if arch is None:
-        return None, 'failed to detect microarchitecture'
+        return None, 'failed to detect microarchitecture in kconfig'
     return arch, 'OK'
 
 
+def detect_arch_sysctl(fname: str, supported_archs: List[str]) -> Tuple[StrOrNone, str]:
+    arch_mapping = {
+        'ARM64': re.compile(r'aarch64|armv[8-9]'), # aarch64 and arm gen 8+ are 64-bit
```

Comment:
As I understand, there is no arch `armv9`, so we can drop it.

Also, the `armv8l` and `armv8b` are the values of `COMPAT_UTS_MACHINE`. The 64-bit kernel shows it to the 32-bit process. Am I right? Could you check that?

Please see the details with examples [here](https://android.stackexchange.com/questions/126927/could-a-64-bit-hardware-device-run-a-32-bit-android-version).